orca-clouddriver/src/main/java/com/netflix/spinnaker/orca/clouddriver/config/CloudDriverConfigurationProperties.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright 2017 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.orca.clouddriver.config;
+
+import java.util.List;
+import java.util.Map;
+import lombok.Data;
+import org.springframework.boot.context.properties.ConfigurationProperties;
+
+@Data
+@ConfigurationProperties
+public class CloudDriverConfigurationProperties {
+
+  @Data
+  public static class BaseUrl {
+    private String baseUrl;
+    private int priority = 1;
+    private Map<String, Object> config;
+
+    BaseUrl(String baseUrl) {
+      this.baseUrl = baseUrl;
+    }
+  }
+
+  @Data
+  public static class MultiBaseUrl {
+    private String baseUrl;
+    private List<BaseUrl> baseUrls;
+  }
+
+  @Data
+  public static class CloudDriver {
+    private String baseUrl;
+    private MultiBaseUrl readonly;
+  }
+
+  private BaseUrl mort;
+  private BaseUrl oort;
+  private BaseUrl kato;
+  private CloudDriver clouddriver;
+
+  String getCloudDriverBaseUrl() {
```

Comment:
this changes the visibility of this method from the groovy default of public, was this intended? it appears to be causing an issue with the aws lambda deployment plugin: https://github.com/spinnaker-plugins/aws-lambda-deployment-plugin-spinnaker/issues/125

```
java.lang.IllegalAccessError: class com.amazon.aws.spinnaker.plugin.lambda.verify.LambdaCacheRefreshTask tried to access method 'java.lang.String com.netflix.spinnaker.orca.clouddriver.config.CloudDriverConfigurationProperties.getCloudDriverBaseUrl()' (com.amazon.aws.spinnaker.plugin.lambda.verify.LambdaCacheRefreshTask is in unnamed module of loader org.pf4j.PluginClassLoader @6a00b08c; com.netflix.spinnaker.orca.clouddriver.config.CloudDriverConfigurationProperties is in unnamed module of loader 'app')
	at com.amazon.aws.spinnaker.plugin.lambda.verify.LambdaCacheRefreshTask.forceCacheRefresh(LambdaCacheRefreshTask.java:93)
```

or is there a more accepted way of getting the cloud driver base url in plugins?

Reply:
it's used a lot in this plugin: https://github.com/spinnaker-plugins/aws-lambda-deployment-plugin-spinnaker/search?q=getCloudDriverBaseUrl

Reply:
https://github.com/spinnaker/orca/pull/4317